src/authenticationinapp/authenticationinapp.cpp

```diff
@@ -48,9 +48,26 @@ void AuthenticationInApp::setState(State state) {
   m_state = state;
   emit stateChanged();
 
-  emit MozillaVPN::instance()->recordGleanEventWithExtraKeys(
-      GleanSample::authenticationInappStep,
-      {{"state", QVariant::fromValue(state).toString()}});
+  if (m_session &&
```

Comment:
In this way, if `m_session` is null, we are unable to send the glean event. What about if we change `setState` like this:
```
void AuthenticationInApp::setState(State state,  AuthenticationInAppSession* session) {
...
```

And in the session destroy slot, you do:
```
AuthenticationInAppSession* session = m_session;
m_session = nullptr;
setState(StateInitializing, session);
```